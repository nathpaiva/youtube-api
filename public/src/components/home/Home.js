'use stict';

import React, { Component } from 'react';
import { connect } from 'react-redux';

import { loadVideosApi, selectVideo, loadMoreVideosApi } from '../../logics/youtubeApi';

import Highlight from '../highlight/Highlight';
import List from '../list/List';

import baseCss from './base.scss';

class Home extends Component {
  state = {
    totalVideos: 4,
    isFetching: false,
  };

  componentDidUpdate(prevProp, prevStatus) {
    if (prevProp.videos.size !== prevStatus.totalVideos) {
      if (this.state.totalVideos == 50) {
        this.setState({ isFetching: true });
        return;
      }

      this.setState({
        isFetching: false
      });
    }
  }

  componentDidMount() {
    this.props.list({ term: '', maxResults: this.state.totalVideos });
  }

  render() {
    return (
      <div className='container'>
        {this.props.selectedVideo.map(video => {
          return <Highlight key={video.id.videoId} selectedVideo={video} />
        })}
        <List
          titleList='+ VÃ­deo'
          videos={this.props.videos}
          videoSelect={video => {
            this.props.changeVideo(video);
          }}
          isFetching={this.state.isFetching}
          moreVideos={(e) => {
            let newTotal = this.state.totalVideos + 10;
            newTotal = newTotal > 50 ? 50 : newTotal
            this.setState({ totalVideos: newTotal, isFetching: true });
            this.props.list({ term: '', maxResults: newTotal });
          }}
          classBtn={'list__more'}
          classList='list__container'
        />
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => {
  return {
    list: (opt) => {
      dispatch(loadVideosApi(opt));
    },
    changeVideo: (video) => {
      dispatch(selectVideo(video));
    }
  }
}

const mapStateToProps = state => {
  return {
    videos: state.listVideosReducer,
    selectedVideo: state.selectVideoReducer,
  }
}

const HomeRedux = connect(mapStateToProps, mapDispatchToProps)(Home);

export default HomeRedux;
