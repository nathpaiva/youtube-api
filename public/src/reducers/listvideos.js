'user strict';

import { List } from 'immutable';

export function listVideosReducer(state = new List(), action) {

  if (action.type === 'LIST') {
    return new List(action.list);
  }

  return state;
}
