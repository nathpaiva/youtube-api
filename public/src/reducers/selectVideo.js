'user strict';

import { List } from 'immutable';

export function selectVideoReducer(state = new List(), action) {

  if (action.type === 'SELECTFIRST') {
    return [action.list[0]];
  }

  if (action.type === 'SELECTONE') {
    return [action.list];
  }


  return state;
}
