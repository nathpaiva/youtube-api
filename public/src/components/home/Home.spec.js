'use stict';

import React from 'react';
import { Route, MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';

import { createStore, applyMiddleware, combineReducers } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { listVideosReducer } from '../../reducers/listvideos';
import { selectVideoReducer } from '../../reducers/selectVideo';

const reducers = combineReducers({ listVideosReducer, selectVideoReducer })
const store = createStore(reducers, applyMiddleware(thunkMiddleware));

import Home from './Home';


describe('#Test Home Component', () => {
  it('#Should Home to mach snapshot - all videos without filter', () => {
    const tree = renderer.create(
      <MemoryRouter initialEntries={['/']}>
        <Route exact path='/' render={routeProps => <Home {...routeProps} store={store} />} />
      </MemoryRouter >
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
