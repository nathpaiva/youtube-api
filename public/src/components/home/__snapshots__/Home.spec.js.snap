// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Test Home Component #Should Home to mach snapshot - all videos without filter 1`] = `
<div
  className="container"
>
  <div>
    <div
      className={undefined}
    >
      <h1>
        + Vídeo
      </h1>
    </div>
    <div
      className={undefined}
    >
      <ul />
      <button
        className={undefined}
        disabled={false}
        onClick={[Function]}
      >
        carregar mais vídeos...
      </button>
    </div>
  </div>
</div>
`;
